core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -458,16 +457,32 @@ public void initialize(URI uri, org.apache.hadoop.conf.Configuration conf) throw
     mStatistics = statistics;
     mUri = URI.create(mAlluxioHeader);
 
-    boolean masterAddIsSameAsDefault = checkMasterAddress();
-
-    if (sInitialized && masterAddIsSameAsDefault) {
+    // Create the master inquire client that we would have after merging the hadoop conf into
+    // Alluxio Configuration.
+    MasterInquireClient.Factory.Config inquireClientConf =
+        MasterInquireClient.Factory.Config.defaults();
+    inquireClientConf.setZookeeperEnabled(conf.getBoolean(PropertyKey.ZOOKEEPER_ENABLED.getName(),
```

Comment:
NIT: is the following a bit more easier to read since you return `this` on setters? leave this up to you  
```java
inquireClientConf
    .setZookeeperEnabled(...)
    .setZookeeperAddress(...)
    ...
```

Reply:
unfortunately we can't do it like that because we only want to override the default values for the fields if they are set in hadoop conf.